platform/uwp/os_uwp.cpp

```diff
@@ -707,6 +709,36 @@ String OS_UWP::get_executable_path() const {
 void OS_UWP::set_icon(const Ref<Image> &p_icon) {
 }
 
+String OS_UWP::get_unique_id() const {
+	//Get the hardware token and read it into an array of bytes
+	HardwareToken ^ token = HardwareIdentification::GetPackageSpecificToken(nullptr);
+	auto hwId = token->Id;
+	DataReader ^ dr = DataReader::FromBuffer(hwId);
+	uint8_t *data = new uint8_t[hwId->Length];
+	dr->ReadBytes(Platform::ArrayReference<uint8_t>(data, hwId->Length));
+
+	//Convert the byte array to hexadecimal
+	String id;
+	char hexStr[3] = "00";
+
+	for (int i = 0; i < hwId->Length; i++) {
+		//Convert next byte to hexadecimal
+		sprintf(hexStr, "%x", (int)data[i]);
+
+		if (data[i] < 16) {
+			hexStr[1] = hexStr[0];
+			hexStr[0] = '0';
+		}
+
+		//Add the next 2 hexits to the ID string
+		id += hexStr;
+	}
+
+	//Free the temporary data array and return the device ID
+	delete[] data;
+	return id;
```

Comment:
```suggestion
	// Get the hardware token and read it into an array of bytes.
	HardwareToken ^ token = HardwareIdentification::GetPackageSpecificToken(nullptr);
	IBuffer hwId = token->Id;
	DataReader ^ dr = DataReader::FromBuffer(hwId);
	uint8_t *data = new uint8_t[hwId->Length];
	dr->ReadBytes(Platform::ArrayReference<uint8_t>(data, hwId->Length));

	// Convert the byte array to hexadecimal.
	String id;
	char hexStr[3] = "00";

	for (int i = 0; i < hwId->Length; i++) {
		// Convert next byte to hexadecimal.
		sprintf(hexStr, "%x", (int)data[i]);

		if (data[i] < 16) {
			hexStr[1] = hexStr[0];
			hexStr[0] = '0';
		}

		// Add the next 2 hexits to the ID string.
		id += hexStr;
	}

	// Free the temporary data array and return the device ID.
	delete[] data;
	return id;
```

We don't use `auto`, and want the types to be explicit to code reviewers. I had to look up the MSDN documentation to know what type this is.

Also style fixes to the comments.

Reply:
One pro of `auto` seems to be that it didn't get the type wrong, I suppose :P 
I ran into an issue with VS 2017, this seems to solve it: https://github.com/godotengine/godot/pull/87288